app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,49 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_order, only: [:payment, :submit_payment, 
+    :show, :edit, :index, :confirm_order]
```

Comment:
`:edit` and `:index` are not actions in this controller, feel free to remove these from the array.